src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+import asyncio
+import json
+import os
+import re
+
+from dataclasses import dataclass
+from lib.clientsession_provider import init_gcp_client_session
+from lib.context import LoggingContext, MetricsContext, get_should_require_valid_certificate
+from lib.credentials import create_token, fetch_dynatrace_api_key, fetch_dynatrace_url
+from lib.metrics import Metric
+from typing import List, Text, Any, Dict
+from lib.configuration import config
+
+logging_context = LoggingContext(None)
+projectID="dynatrace-gcp-extension"
+discovered_resource_type="cloud_function"
+
+
+
+@dataclass(frozen=True)
+class MetricMetadata:
+    """Describes singular GCP service to ingest data from."""
+    # IMPORTANT! this object is only for one combination of object/featureSet!
+    # If you have 2 featureSets enabled for some service you will have 2 such objects
+
+    display_name: Text
+    description: Text
+    unit: Text
+    tags: List[Text]
+    dimensions: List[Dict[Text,Text]]
+    metric_properties: Dict[Text,Any]
+    schema_id: Text
+    scope: Text
+
+
+    def __init__(self, **kwargs):
+        object.__setattr__(self, "displayName", kwargs.get("displayName", ""))
```

Comment:
I think python constructor should look like this
![image](https://github.com/dynatrace-oss/dynatrace-gcp-monitor/assets/16317514/19485563-c198-4f09-a659-24ba6d9be08c)

I know we have this weird setattr in some places, we have to remove them :D But for new code please make it clean